### PR TITLE
Dapp add devote

### PR DESCRIPTION
**Pull request by 0xExp-po**
First fixed bugs in register and update project page UI
- check the valid input of project info before register
- after register display correct project info in project page
- update the UI based on your feedback

After that I'll keep working to implement devote function

### CODE REVIEW

File: dapp/src/components/DonateModal.astro [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+            </div>
+          </div>
+        </span>
+        <div class="flex items-center relative">
+          <span class="absolute left-2 sm:left-3 sm:py-0.5 pr-1 border-r-2 border-zinc-600">XLM</span>
+          <input 
+            id="tip-amount-input" 
+            type="number" 
+            class="w-28 sm:w-36 flex-1 border rounded-lg pl-12 sm:pl-14 pr-3 py-0.5 sm:py-1 text-sm sm:text-base text-right" 
+            value=
```
Comment (tupui) [inline dapp/src/components/DonateModal.astro:79]: 0 as default and in that case should not trigger any transaction.
```suggestion
            value=0
```

**Comment by 0xExp-po** [comment]
Yes, the soroban domain contract may be updated now.
according to the docs, I've implemented donate function but I can't read the domain address
so, I tested payment with "tansu_domain_address"

**Comment by 0xExp-po** [comment]
I've fixed and updated some things based on your feedback

**Review by tupui — APPROVED** [review]
Let's get this in! Thanks again 🚀

**Pull request merged**